src/components/calcite-input/calcite-input.tsx

```diff
@@ -248,68 +294,95 @@ export class CalciteInput {
 
     const suffixText = <div class="calcite-input-suffix">{this.suffixText}</div>;
 
-    const childEl =
-      this.childElType !== "textarea" ? (
-        <input
-          {...attributes}
-          autofocus={this.autofocus ? true : null}
-          disabled={this.disabled ? true : null}
-          max={this.maxString}
-          min={this.minString}
-          onBlur={this.inputBlurHandler}
-          onFocus={this.inputFocusHandler}
-          onInput={this.inputInputHandler}
-          placeholder={this.placeholder || ""}
-          ref={(el) => (this.childEl = el)}
-          required={this.required ? true : null}
-          step={this.stepString}
-          tabIndex={this.disabled ? -1 : null}
-          type={this.type}
-          value={this.value}
+    const childEl = [
+      <this.childElType
+        {...attributes}
+        autofocus={this.autofocus ? true : null}
+        disabled={this.disabled ? true : null}
+        max={this.maxString}
+        min={this.minString}
+        onBlur={this.inputBlurHandler}
+        onFocus={this.inputFocusHandler}
+        onInput={this.inputInputHandler}
+        placeholder={this.placeholder || ""}
+        ref={(el) => (this.childEl = el)}
+        required={this.required ? true : null}
+        step={this.stepString}
+        tabIndex={this.disabled ? -1 : null}
+        type={this.type}
+        value={this.value}
+      >
+        {this.value}
+      </this.childElType>,
+      this.isTextarea ? (
+        <div class="calcite-input-resize-icon-wrapper">
+          <calcite-icon icon="chevron-down" scale="s" />
+        </div>
+      ) : null
+    ];
+
+    const labelEl = (
```

Comment:
Sorry if I'm missing something here, why do we need to create a label inside the component? Users would likely have wrapped the component already like:

```
<calcite-label>
My editable thing
<calcite-input editableInput />
</calcite-label>
```

If it's to display the "non editing state" (sorry can't get that to display on demo), maybe just a div? Not sure about the semantics of replacing with the hidden input.

Reply:
Can use another tag, but calcite-label made it super easy to apply all the passed in style props (i.e. scale, theme, etc) without having to write custom CSS styles for another tag.

Reply:
That's fair, just need to check if in the above example where a user wraps the input in a calcite-label (or native label), if nesting the input inside a second label still works for a11y :)